packages/json-table-schema-visualizer/src/providers/TableDetailLevelProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  createContext,
+  useCallback,
+  useEffect,
+  useState,
+  type ReactNode,
+} from "react";
+
+import {
+  type TableDetailLevelValue,
+  TableDetailLevel,
+} from "@/types/tabelDetailLevel";
+import { detailLevelStore } from "@/stores/detailLevelStore";
+
+export const TableDetailLevelContext = createContext<TableDetailLevelValue>({
+  detailLevel: TableDetailLevel.FullDetails,
+  next() {},
+});
+
+interface TabelLevelDetailProviderProps {
+  children: ReactNode;
+  level?: TableDetailLevel;
+}
+
+const TabelLevelDetailProvider = ({
+  children,
+}: TabelLevelDetailProviderProps) => {
+  const [state, setState] = useState(detailLevelStore.getCurrentDetailLevel());
```

Comment:
A title suggestion
```suggestion
  const [state, setState] = useState( () => detailLevelStore.getCurrentDetailLevel() );
```
The advantage is that the `detailLevelStore.getCurrentDetailLevel` will be called only once. At the first rendering React will call it to get the initial value. For the subsequent rendering, React will no longer call it.